api/provider/aws/common.go

```diff
@@ -406,20 +405,21 @@ func waitUntilSGDeletedFN(ec2api ec2iface.EC2API, securityGroupName string) func
 		if err != nil {
 			log.Printf("[DEBUG] could not delete security group due to %s", err.Error())
 			if err, ok := err.(awserr.Error); ok && err.Code() == "DependencyViolation" {
-				return true, nil
+				return true
 			}
 
-			return false, err
+			*sgError = err
```

Comment:
this just seems very confusing - not sure why you don't just use the same pattern you have elsewhere?